### DISCUSSION TITLE
Adjust sampling frequency

### DISCUSSION THREAD

**Discussion by apaaris** [theOehrly/Fast-F1#258] [Q&A] [answered]
Dear everyone, 
I wanted to have a way to decrease/increase the frequency of the measurements, I saw that it's possible to do it with the following attribute of the Telemetry class: ```TELEMETRY_FREQUENCY```

```python
 t1 = fastest1.get_telemetry().add_distance()
 t1.TELEMETRY_FREQUENCY = 100
```

Something like that would not give errors, however I suppose that the data needs to be reloaded in some way       
However I do not know how to set it properly, is there a way to put it as a global variable at the beginning of a script and having it used as a setting everywhere possible?

Thank you a lot

**Comment by theOehrly** [comment, marked as answer]
At the beginning of the script, you can do
```
from fastf1.core import Telemetry

Telemetry.TELEMETRY_FREQUENCY = 100

...

t1 = fastest1.get_telemetry()
```

HOWEVER, this will absolutely not change the measurement frequency. It will simply **upsample** the data to a higher frequency. But this might actually **decrease** the accuracy of the data, because upsampling means creating values that were not previously there so it's kind of "guesswork".

**Reply by apaaris** [reply]
Thanks a lot! This solved my problem!
